### PR TITLE
Rename scaffold-to-generate

### PR DESCRIPTION
**Pull request by pepicrft**
Resolves https://github.com/Shopify/shopify-cli-planning/issues/325

### WHY are these changes introduced?
As concluded by @MeredithCastile, `generate` is a more suitable naming for the current `scaffold` action.

### WHAT is this pull request doing?
- [x] Add the new `generate` command.
- [x]  Adjust the old command to indicate that's deprecated.
- [x]  Adjust analytics to continue sending the command as `scaffold`.
- [x]  Update templates to rename `scaffold` to `generate` in their `package.json`s
  - [x] Node: https://github.com/Shopify/shopify-app-template-node/pull/1028
  - [x] PHP: https://github.com/Shopify/shopify-app-template-php/pull/328 
  - [x] Ruby: https://github.com/Shopify/shopify-app-template-ruby/pull/36 

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [x] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### How to test your changes?
Try to generate a new extension in the fixture app using the new `generate` command.

### CODE REVIEW

File: packages/app/src/cli/commands/app/scaffold/extension.ts [hunk 1]
```diff
@@ -1,222 +1,20 @@
...
-    outputTokens.push(
-      output.content`\n  To find your extension, remember to ${output.token.genericShellCommand(
-        output.content`cd ${output.token.path(extensionDirectory)}`,
-      )}`.value,
+    output.warn(
+      output.content`The command ${output.token.genericShellCommand(
+        `scaffold`,
+      )} has been deprecated in favor of ${output.token.genericShellCommand(
+        `generate`,
+      )} and will be deleted in the next major version.
```
Comment (gonzaloriestra) [inline packages/app/src/cli/commands/app/scaffold/extension.ts:13]: In 4.x? If we are not still sure about when, maybe we can just be less explicit for now:

```suggestion
      )} and will be eventually deleted.
```

File: packages/app/src/cli/commands/app/scaffold/extension.ts [hunk 2]
```diff
@@ -1,222 +1,20 @@
...
-      hidden: false,
-      description: 'Choose a starting template for your extension, where applicable',
-      options: ['vanilla-js', 'react', 'typescript', 'typescript-react', 'wasm', 'rust'],
-      env: 'SHOPIFY_FLAG_TEMPLATE',
-    }),
-  }
-
-  static args = [{name: 'file'}]
+class AppScaffoldExtension extends AppGenerateExtension {
+  static description = 'Scaffold an Extension (deprecated in favor of generate)'
```
Comment (gonzaloriestra) [inline packages/app/src/cli/commands/app/scaffold/extension.ts:5]: Wouldn't it be better to hide the scaffold command in the help?

**Review by gonzaloriestra — APPROVED** [review]

File: packages/app/src/cli/commands/app/generate/extension.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+      description: 'Choose a starting template for your extension, where applicable',
+      options: ['vanilla-js', 'react', 'typescript', 'typescript-react', 'wasm', 'rust'],
+      env: 'SHOPIFY_FLAG_TEMPLATE',
+    }),
+  }
+
+  static args = [{name: 'file'}]
+
+  public static analyticsNameOverride(): string | undefined {
+    return 'scaffold'
```
[the remark this answers is not in the document]
Comment (shauns) [reply packages/app/src/cli/commands/app/generate/extension.ts]: Please no on this one. There is no going back and updating the old data, its an immutable log. Bear in mind that with this approach, users will continue to produce 'scaffold' events (if they don't upgrade their CLI -- a significant %), and so we will _always_ have to combine scaffold and generate events into one set. This becomes a permanent tax on analysis (it is not obvious at all that you have to do this) and we need self-service analysis to be as foolproof as possible.

Or to put it another way, we are going to get asked what the difference is between "generate" and "scaffold" in the logs, forever!

File: packages/app/src/cli/commands/app/scaffold/extension.ts (hunk 1 above)
Comment (shauns) [reply packages/app/src/cli/commands/app/scaffold/extension.ts:13]: We won't have a 4.0 for quite some time.

File: packages/app/src/cli/commands/app/generate/extension.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (gonzaloriestra) [reply packages/app/src/cli/commands/app/generate/extension.ts]: The original approach has also some problems:
- We will send `scaffold` forever, even after removing the command. We are going to get asked about that as well.
- We will never know how many people use scaffold vs generate, useful to decide when to remove the command.

But I don't like any solution actually, up to you! 🤷

[the remark this answers is not in the document]
Comment (pepicrft) [reply packages/app/src/cli/commands/app/generate/extension.ts]: I'll open a follow-up PR to send the event as `scaffold`.

**Pull request merged**